### PR TITLE
Update README.md to direct to FF release notes

### PR DESCRIPTION
**Pull request by amcisaac**
Added a link to our org website that highlights release notes, github links, and dois of current force fields.

### CODE REVIEW

**Comment by amcisaac** [comment]
One thing to note is that the force field website page contains links to github, but it's a link to the `openff-forcefields` release, not a direct link to the fitting repo. I still feel like that's fine, and users can go to the `openff-forcefields` repo to find the fitting repo easily after my most recent update. Let me know what you think.

**Comment by amcisaac** [comment]
@j-wags could you please merge this if it looks good? Sorry I don't have write access.

**Review by j-wags — APPROVED** [review]

**Pull request merged**
